src/extra/libs/png/lv_png.h

```diff
@@ -19,6 +19,7 @@ extern "C" {
 /*********************
  *      DEFINES
  *********************/
+#define LV_PNG_ID   0x06
```

Comment:
It seems a little bit error prone to define magic numbers manually. What about using strings, like `const char * magic_id = "PNG"`?

Reply:
I've tried to optimize the structure of the decoder context, so it fits with a 16 bits alignment. The magic id is thus 8 bits, followed by a 8 bits caps. I've used leetspeak to decide the format ID (by ignoring the letter that does not map to a hex number), but I agree it's dumb.

I don't like using strings here, since comparing string is costly while comparing a number is essentially free.
In #2926 I propose another option, to use the index in the decoder's linked list instead (or the hash map) so we can directly jump to the right decoder instead of trying them one by one.

Reply:
With #2926 `strcmp` were used only once on setting a new image. So it shouldn't slow down the system in a measurable way.  I think having less error prone IDs is worth more than saved a few short `strcmp`s in `lv_img_set_src`.

> I propose another option, to use the index in the decoder's linked list instead

An index in  the linked list could be dangerous as the linked list can change in runtime.